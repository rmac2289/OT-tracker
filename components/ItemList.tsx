import React, { useEffect, useState } from "react";
import { StyleSheet, View, Text, ScrollView, Platform } from "react-native";
import * as Calendar from "expo-calendar";
import CalendarItem from "./CalendarItem";
import { calendar_id } from "../config";
import { formatDate } from "../lib/formatDate";

const ItemList = () => {
  const [ot, setOt] = useState<Calendar.Event[]>([]);
  useEffect(() => {
    (async () => {
      const { status } = await Calendar.requestCalendarPermissionsAsync();
      if (status === "granted") {
        const calendars = await Calendar.getEventsAsync(
          [calendar_id],
          new Date(),
          new Date(2021, 8)
        );
        let events = { calendars };

        let filtered = events.calendars.filter(
          (event) => event.title === "Ross Forced OT"
        );
        setOt(filtered);
      }
    })();
  }, []);

  return (
    <ScrollView style={styles.scrollview}>
      {ot.map((event: Calendar.Event, index: number) => {
        return (
          <CalendarItem
            key={index}
            dateDay={formatDate(new Date(event.startDate), 0, 3)}
            dateMonth={formatDate(new Date(event.startDate), 4, 10)}
            dateTimeStart={formatDate(new Date(event.startDate), 16, 21)}
            dateTimeEnd={formatDate(new Date(event.endDate), 16, 21)}
          />
        );
      })}
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  text: {
    fontSize: 20,
  },
  scrollview: {
    width: "100%",
  },
});

export default ItemList;
