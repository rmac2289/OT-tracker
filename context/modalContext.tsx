import React, { createContext, useState } from "react";

interface Props {}

export const ModalContext = createContext<any>(null);

export const ModalContextProvider = ({
  children,
}: {
  children: JSX.Element;
}) => {
  const [fadeBackground, setFadeBackground] = useState(false);

  return (
    <ModalContext.Provider value={[fadeBackground, setFadeBackground]}>
      {children}
    </ModalContext.Provider>
  );
};
