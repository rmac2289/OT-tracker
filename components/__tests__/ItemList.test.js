import React from "react";
import renderer from "react-test-renderer";
import { ModalContextProvider } from "../../context/modalContext";
import ItemList from "../ItemList";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <ModalContextProvider>
        <ItemList />
      </ModalContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
