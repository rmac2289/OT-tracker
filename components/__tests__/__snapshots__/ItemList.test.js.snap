// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "height": 100,
          "left": "50%",
          "marginLeft": -50,
          "marginTop": 100,
          "position": "absolute",
          "top": "50%",
          "width": 100,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#a4545f"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</RCTScrollView>
`;
